src/the_way/mod.rs

```diff
@@ -86,13 +86,19 @@ impl TheWay {
             }
             TheWayCLI::View { index } => self.view(*index),
             TheWayCLI::List { filters } => self.list(filters),
-            TheWayCLI::Import { file } => {
+            TheWayCLI::Import { file, gist_url } => {
                 let mut num = 0;
-                for mut snippet in self.import(file.as_deref())? {
-                    snippet.index = self.get_current_snippet_index()? + 1;
-                    self.add_snippet(&snippet)?;
-                    self.increment_snippet_index()?;
-                    num += 1;
+                if let Some(gist_url) = gist_url {
+                    let gist_url = gist_url.clone();
```

Comment:
Hmm, I understand the need for this clone with the current setup since we're borrowing `self` immutably in line 71, but just realized that `cli` doesn't need to be a struct attribute at all. This would mean removing `cli` from `TheWay` struct,  adding it as an input to the run function (`fn run(&mut self, cli: TheWayCLI) -> color_eyre::Result<()>`), changing `match &self.cli` to `match cli`, removing the clones and basically letting clippy tell you how to fix the rest. Let me know if you're up for trying this out, if not I'll merge this with the clone and change it later.

Reply:
I tried to address this in https://github.com/out-of-cheese-error/the-way/pull/79/commits/efe4cc9dd14411ac2e7ebe8af1e05b998b6d59cc.